subprojects/core/src/main/java/org/gradle/api/internal/attributes/AbstractAttributeContainer.java

```diff
@@ -51,4 +53,25 @@ public AttributeContainer getAttributes() {
     public abstract boolean equals(Object o);
     @Override
     public abstract int hashCode();
+
+    /**
+     * Verifies that the given key is a valid attribute request.
+     * <p>
+     * Requesting null keys is not allowed, and is deprecated.
+     *
+     * @param key the requested attribute to validate
+     * @return {@code true} is valid request; {@code false} otherwise
+     */
+    protected boolean isValidAttributeRequest(@Nullable Attribute<?> key) {
+        if (key == null) {
+            DeprecationLogger.deprecateInvocation("getAttribute with a null key")
```

Comment:
I think this is intended to deprecate a method call, but this is a behavior.

I would use `deprecateBehaviour("Retrieving attribute with a null key")` and avoid making this look like a method is going to be removed.